src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -48,6 +48,9 @@ CHIP_ERROR CASEServer::ListenForSessionEstablishment(Messaging::ExchangeManager
     // Set up the group state provider that persists across all handshakes.
     GetSession().SetGroupDataProvider(mGroupDataProvider);
 
+    ChipLogProgress(Inet, "CASE Server enabling CASE session setups");
+    mExchangeManager->RegisterUnsolicitedMessageHandlerForType(Protocols::SecureChannel::MsgType::CASE_Sigma1, this);
```

Comment:
Previous code would have UMH added on:
   - Listen
   - OnSessionEstablishmentError
   - OnEsssionEstablished

Now we would only listen on "Listen" calls but we would not re-register in case of CASE errors or after a session finshes establishment. Could you explain in the PR summary why this is ok and expected?